cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -133,7 +133,6 @@ func CreateStaticPodFiles(manifestDir, patchesDir string, cfg *kubeadmapi.Cluste
 func getAPIServerCommand(cfg *kubeadmapi.ClusterConfiguration, localAPIEndpoint *kubeadmapi.APIEndpoint) []string {
 	defaultArguments := map[string]string{
 		"advertise-address":               localAPIEndpoint.AdvertiseAddress,
-		"insecure-port":                   "0",
```

Comment:
if the flags for KCM and scheduler are removed too, i guess we should remove them from the kubeadm manifests:
https://github.com/kubernetes/kubernetes/blob/c27c7a545b3ae2bf7e9a85830b9b83052a5ff4f9/cmd/kubeadm/app/phases/controlplane/manifests.go#L325

https://github.com/kubernetes/kubernetes/blob/c27c7a545b3ae2bf7e9a85830b9b83052a5ff4f9/cmd/kubeadm/app/phases/controlplane/manifests.go#L396

Reply:
does kubeadm construct manifests for any older versions? we want to be absolutely positive we don't omit the flag to an old binary

Reply:
looks like i forgot about the skew. kubeadm supports deploying X and X-1 of the control-plane.

this means that the version in cfg.KubernetesVersion has to be checked.
if this version of kubeadm is deploying a component version (cfg.KubernetesVersion) that no longer has the flags, the flags have to be removed from the manifests. for older versions they have to be preserved.